packages/next-tweet/src/tweet-video.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+
+import { FC, useState } from 'react';
+import Image from 'next/image';
+import { getMediaUrl } from './utils';
+import s from './tweet-media.module.css';
+import { VideoProps } from './tweet-media';
+
+export const VideoPlayer: FC<VideoProps> = ({
+  media, priority,
+}) => {
+  const [displayVideo, setDisplayVideo] = useState(false);
+  const sortedVideos = media.video_info.variants.sort((a, b) => {
+    return (a.bitrate ?? 0) < (b.bitrate ?? 0) ? 1 : 0;
+  });
+  if (displayVideo) {
+    return (
+      <div>
+        <video
+          className={s.image}
+          controls
+          autoPlay
+          muted
+          preload={priority ? 'auto' : 'metadata'}
```

Comment:
This doesn't do anything because when `autoPlay` is set the `preload` attribute is ignored. Also the video tag is never seen by the browser because it's only added after an interaction.